scope

```diff
@@ -128,18 +148,33 @@ EOF
         ;;
 
     launch)
-        check_not_running $SCOPE_CONTAINER_NAME $SCOPE_IMAGE_NAME
-        docker rm -f $SCOPE_CONTAINER_NAME >/dev/null 2>&1 || true
-
-        CONTAINER=$($(launch_command "$@"))
-        echo $CONTAINER
+        if ! check_docker_for_mac || check_probe_only ; then
+            check_not_running $SCOPE_CONTAINER_NAME $SCOPE_IMAGE_NAME
+            docker rm -f $SCOPE_CONTAINER_NAME >/dev/null 2>&1 || true
+            CONTAINER=$($(launch_command "$@"))
+            echo $CONTAINER
+        elif check_docker_for_mac ; then
+            check_not_running $SCOPE_APP_CONTAINER_NAME $SCOPE_IMAGE_NAME
+            docker rm -f $SCOPE_APP_CONTAINER_NAME >/dev/null 2>&1 || true
+            CONTAINER=$($(launch_app_command "$@"))
+            echo $CONTAINER
+            app_ip=$(docker inspect -f '{{.NetworkSettings.IPAddress}}' "${CONTAINER}")
+            check_not_running $SCOPE_CONTAINER_NAME $SCOPE_IMAGE_NAME
+            docker rm -f $SCOPE_CONTAINER_NAME >/dev/null 2>&1 || true
+            CONTAINER=$($(launch_command --no-app "${app_ip}:4040" "$@"))
```

Comment:
Does `"$@"` need to come before `"${app_ip}:4040"` as `"$@"` might container flags.